runtime/near-vm-logic/src/alt_bn128.rs

```diff
@@ -47,20 +28,31 @@ pub fn alt_bn128_g1_multiexp_sublinear_complexity_estimate(n_bytes: u64, discoun
 
     // A+C*n/(log2(n)+4) - B*n - discount
 
-    let n = (n_bytes + MULTIEXP_ITEM_SIZE - 1) / MULTIEXP_ITEM_SIZE;
-    let mut res = A + if n == 0 {
-        0
-    } else {
-        // set linear complexity growth for n > 32768
-        let l = std::cmp::min(ilog2(n), 15);
-        (n * (l + 3) + (1 << (1 + l))) * C / ((l + 4) * (l + 5))
-    };
-
-    if res < B * n + discount {
-        return 0;
-    }
-    res -= B * n + discount;
-    res
+    let n = (n_bytes - 1).checked_add(MULTIEXP_ITEM_SIZE).ok_or(HostError::IntegerOverflow)?
+        / MULTIEXP_ITEM_SIZE;
```

Comment:
This will underflow if `n_bytes == 0`. This also assumes that (normally) `n_bytes` is a multiple of the item size, but it actually includes extra 4 bytes for the number of items, so `n` will be one greater than the number of items.